ocaml/parsing/builtin_attributes.ml

```diff
@@ -27,14 +27,40 @@ end)
 let unused_attrs = Attribute_table.create 128
 let mark_used t = Attribute_table.remove unused_attrs t
 
+let compare_location : Location.t -> Location.t -> int =
+  fun { loc_start = { pos_fname = loc_start_pos_fname_1
+                    ; pos_lnum = _ (* only need character number *)
+                    ; pos_bol = _ (* only need character number *)
+                    ; pos_cnum = loc_start_pos_cnum_1 }
+      ; loc_end = { pos_fname = _ (* locations should live in one file *)
```

Comment:
Ppxes can generate arbitrary location with arbitrary frame.

Reply:
Good point.  But imagine we're comparing two locations which both start in the same file but end in different files.  How should we incorporate that information into the ordering?  What does such a location even mean?

I suppose we can compare the end file names alphabetically as well, and incorporate that into the ordering.  I'd argue against bothering with that if this comparison were used in code on the critical path of type checking, just for efficiency reasons.  But as it's only used for warnings it could be OK.

@antalsz I'm fine either way.

Reply:
This is an edge case that may not happen often in practice, but interleaving warning with different filenames will be confusing

Reply:
@hhugo I changed things around to use a full order; does this look better to you?